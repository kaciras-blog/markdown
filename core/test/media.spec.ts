import { expect, it } from "vitest";
import MarkdownIt from "markdown-it";
import clientMediaPlugin from "../src/web/media.ts";

const md = new MarkdownIt();
md.use(clientMediaPlugin);

it("should output minimized HTML", () => {
	expect(md.render("@video[](/file.mp4)")).toMatchSnapshot();
});

it("should replace builtin image renderer", () => {
	expect(md.render("![label](/file.png)")).toMatchSnapshot();
});

it("should add aspect ratio to style", () => {
	expect(md.render("![label](/file.png?vw=11&vh=22)")).toMatchSnapshot();
});
