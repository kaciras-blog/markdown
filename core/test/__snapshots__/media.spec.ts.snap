// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should add aspect ratio to style 1`] = `
"<p><span class=\\"center-wrapper\\"><a style='--width:11px;--aspect:11/22' href=\\"/file.png?vw=11&vh=22\\" target=\\"_blank\\" rel=\\"noopener,nofollow\\"><img data-src=\\"/file.png?vw=11&vh=22\\" alt=\\"label\\" class=\\"md-img\\" crossorigin></a><span class=\\"md-alt\\">label</span></span></p>
"
`;

exports[`should add label section to audio 1`] = `"<p class=\\"center-wrapper\\"><audio controls src=\\"bar.opus\\" crossorigin></audio><span class=\\"md-alt\\">foo</span></p>"`;

exports[`should output minimized HTML 1`] = `"<p class=\\"center-wrapper\\"><video class=\\"md-video\\" controls crossorigin poster=\\"\\" data-src=\\"/file.mp4\\"></video></p>"`;

exports[`should replace builtin image renderer 1`] = `
"<p><span class=\\"center-wrapper\\"><a  href=\\"/file.png\\" target=\\"_blank\\" rel=\\"noopener,nofollow\\"><img data-src=\\"/file.png\\" alt=\\"label\\" class=\\"md-img\\" crossorigin></a><span class=\\"md-alt\\">label</span></span></p>
"
`;
